compiler/src/iree/compiler/Codegen/Utils/Utils.cpp

```diff
@@ -597,5 +598,35 @@ linalg::LinalgLoopDistributionOptions getIREELinalgLoopDistributionOptions() {
                std::function<linalg::ProcInfo(OpBuilder &, Location)>>()};
 }
 
+void replaceMemrefUsesAndPropagateType(Operation *oldOp, Value val,
```

Comment:
In general I'm -1 on such utility functions that recurse and rewrite operations using an `OpBuilder`. First recursion is not great (see discussion on https://discourse.llvm.org/t/rfc-update-to-mlir-developer-policy-on-recursion/62235), second using `Operation::erase()` methods directly are incompatible with pattern rewriters (these expect erase to go through `RewriterBase::erase()` methods. Its a difference in style, but having this in the `Utils` folder encourages uses everywhere, and I am not for that.

Reply:
Is the problem where it is or the fact that this erase or that it is in Utils? This kind of transformation cannot be done in a Rewriter as it touches ops that are not the root. For recursion I can change it if you think this is a concern however in general it shouldn't recurse very deeply.
Do you have a preference for a different location to share this?

Reply:
Recursion is a bit of a problem, the bigger problem is the `erase`. This utility function is unsafe to use in a pattern rewriter. So then it is not really a utility method cause it can only be used in walks.
I really wish we could just `memref.cast` here to fix up the return type and that propagates. I know it doesnt work in some cases, but I thought it should work in this case....

Reply:
doing a memref.cast doesn't work in this case as the map generated by the subview is not an identity. Should be it moved to a different file then? If so which one?

Reply:
Now that it is in, lets leave it there. I would have suggested keeping it in `LLVMGPU/GPUUtils.h` or something till we need it somewhere else. Now that it is in Utils, chances are even if we move it out we might move it back into Utils. That back and forth is not worth it...

Reply:
ok